src/components/repository-browser.js

```diff
@@ -264,16 +269,30 @@ class RepositoryBrowser extends React.Component {
     this.increaseCurrentUploadCount();
 
     const { cdnApiClient } = this.props;
-    const prefix = pathHelper.getPrefix(pathSegments);
+    const { currentPathSegments, selectedRowKeys } = this.state;
+    const prefix = pathHelper.getPrefix(currentPathSegments);
 
     await cdnApiClient.uploadFiles(files, prefix, { onProgress });
 
     this.decreaseCurrentUploadCount();
 
-    const { currentPathSegments, selectedRowKeys } = this.state;
     await this.refreshFiles(currentPathSegments, selectedRowKeys);
   }
 
+  async handleDeleteFile(fileName) {
+    const { cdnApiClient, onSelectionChanged } = this.props;
+    const { currentPathSegments, selectedRowKeys } = this.state;
+    const prefix = pathHelper.getPrefix(currentPathSegments);
+    const objectName = `${prefix}${fileName}`;
+
+    await cdnApiClient.deleteFile(prefix, fileName);
+    if (selectedRowKeys.includes(objectName)) {
```

Comment:
This is actually `if (selectedRowKeys === fileName)`, or?

Reply:
That's what I thought as well, but that wasn't the case.